tests/testMatrix.cpp

```diff
@@ -116,212 +192,95 @@ int main(int argc, char** argv)
+
+  if(rank == 0)
   {
-    if (rank==0)
-      std::cout << "\nTesting hiopMatrixDenseRowMajor ...\n";
-    // Matrix dimensions denoted by subscript
-    // Distributed matrices (only distributed by columns):
-    hiop::hiopMatrixDenseRowMajor A_kxm(K_local, M_global, m_partition, comm);
-    hiop::hiopMatrixDenseRowMajor A_kxn(K_local, N_global, n_partition, comm);
-    hiop::hiopMatrixDenseRowMajor A_mxk(M_local, K_global, k_partition, comm);
-    hiop::hiopMatrixDenseRowMajor A_mxn(M_local, N_global, n_partition, comm);
-    hiop::hiopMatrixDenseRowMajor A_nxm(N_local, M_global, m_partition, comm);
-    // Try factory instead of constructor
-    hiop::hiopMatrixDense* B_mxn = 
-      hiop::LinearAlgebraFactory::createMatrixDense(M_local, N_global, n_partition, comm);
-    hiop::hiopMatrixDense* A_mxn_extra_row =
-      hiop::LinearAlgebraFactory::createMatrixDense(M_local, N_global, n_partition, comm, M_local+1);
-
-    // Non-distributed matrices:
-    hiop::hiopMatrixDenseRowMajor A_mxk_nodist(M_local, K_local);
-    // Try factory instead of constructor
-    hiop::hiopMatrixDense* A_mxm_nodist =
-      hiop::LinearAlgebraFactory::createMatrixDense(M_local, M_local);
-    hiop::hiopMatrixDenseRowMajor A_kxn_nodist(K_local, N_local);
-    hiop::hiopMatrixDenseRowMajor A_kxm_nodist(K_local, M_local);
-    hiop::hiopMatrixDenseRowMajor A_mxn_nodist(M_local, N_local);
-    hiop::hiopMatrixDenseRowMajor A_nxn_nodist(N_local, N_local);
-
-    // Vectors with shape of the form:
-    // x_<size>_[non-distributed]
-    //
-    // Distributed vectors:
-    hiop::hiopVectorPar x_n(N_global, n_partition, comm);
-
-    // Non-distributed vectors
-    hiop::hiopVectorPar x_n_nodist(N_local);
-    hiop::hiopVectorPar x_m_nodist(M_local);
-
-    hiop::tests::MatrixTestsDense test;
-
-    fail += test.matrixSetToZero(A_mxn, rank);
-    fail += test.matrixSetToConstant(A_mxn, rank);
-    fail += test.matrixTimesVec(A_mxn, x_m_nodist, x_n, rank);
-    fail += test.matrixTransTimesVec(A_mxn, x_m_nodist, x_n, rank);
-
-    if(rank == 0)
-    {
-      // These functions are only meant to be called locally
-
-      fail += test.matrixTimesMat(A_mxk_nodist, A_kxn_nodist, A_mxn_nodist);
-      fail += test.matrixAddDiagonal(A_nxn_nodist, x_n_nodist);
-      fail += test.matrixAddSubDiagonal(A_nxn_nodist, x_m_nodist);
-      //fail += test.matrixAddToSymDenseMatrixUpperTriangle(A_nxn_nodist, A_mxk_nodist);
-      fail += test.matrixTransAddToSymDenseMatrixUpperTriangle(A_nxn_nodist, A_kxm_nodist);
-      fail += test.matrixAddUpperTriangleToSymDenseMatrixUpperTriangle(A_nxn_nodist, *A_mxm_nodist);
+    // These methods are local
+    fail += test.matrixTimesMat(*A_mxk_nodist, *A_kxn_nodist, *A_mxn_nodist);
+    fail += test.matrixAddDiagonal(*A_nxn_nodist, *x_n_nodist);
+    fail += test.matrixAddSubDiagonal(*A_nxn_nodist, *x_m_nodist);
+    //fail += test.matrixAddToSymDenseMatrixUpperTriangle(*A_nxn_nodist, *A_mxk_nodist);
```

Comment:
Do we want to be testing `matrixAddToSymDenseMatrixUpperTriangle`? Currently, the test is not implemented

Reply:
> Some choices I made may not be the best if we intend to test distributed matrices under the same testing framework as dense matrices. If this is the case, I can reverse some of the changes that I highlighted that could be problematic.

Yes, please reverse those changes. The objective of this PR is to simplify unit test helper methods, but not at the expense of reducing the functionality of unit tests.